src/gretel_trainer/models.py

```diff
@@ -1,9 +1,28 @@
 import logging
 from typing import Union
 
+import pandas as pd
```

Comment:
This is imported only for the type hint in `determine_best_model`... I'm open to the argument that this isn't necessary and we just don't type hint that function

Reply:
We've been using this pattern in cases where import is only used for type hinting and I think that's what would work well here as well.

```
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    import pandas as pd
```